llvm/lib/IR/StructuralHash.cpp

```diff
@@ -100,8 +233,20 @@ class StructuralHashImpl {
     if (const auto *ComparisonInstruction = dyn_cast<CmpInst>(&Inst))
       Hashes.emplace_back(ComparisonInstruction->getPredicate());
 
-    for (const auto &Op : Inst.operands())
-      Hashes.emplace_back(hashOperand(Op));
+    unsigned InstIdx = 0;
+    if (IndexInstruction) {
+      InstIdx = IndexInstruction->size();
+      IndexInstruction->insert({InstIdx, const_cast<Instruction *>(&Inst)});
```

Comment:
I think otherwise `InstrIdx` will be wrong if the instruction was not actually inserted into the map.

```suggestion
      auto [It, WasInserted] = IndexInstruction->try_emplace(IndexInstruction->size(), const_cast<Instruction *>(&Inst));
      InstIdx = It->first;
```

Reply:
Instruction is inserted once by design in this pass. In fact, this map `IndexInstruction` itself can't catch the duplication as the key is `index`, not `Instruction *`.  Anyhow, replaced `insert` by `trace_emplace` for efficiency.